### PR TITLE
moved results_to_compute to [train|infer]_input classes

### PR DESCRIPTION
**Pull request by makart19**
moved functionality for assignment of results_to_compute to train|infer input classes

### CODE REVIEW

File: examples/oneapi/dpc/source/decision_forest/df_reg_dense_batch.cpp [hunk 1]
```diff
@@ -69,18 +69,17 @@ void run(sycl::queue &queue) {
...
   const auto df_infer_desc =
       df::descriptor<float, df::task::regression, df::method::dense>();
 
   try {
     const auto result_train =
-        dal::train(queue, df_train_desc, x_train_table, y_train_table);
+        dal::train(queue, df_train_desc, x_train_table, y_train_table,
+              df::train_result_to_compute::compute_out_of_bag_error |
+              df::train_result_to_compute::
+                  compute_out_of_bag_error_per_observation);
```
Comment (rlnx) [inline examples/oneapi/dpc/source/decision_forest/df_reg_dense_batch.cpp:82]: Shouldn't it be on one line? Is it clang-format or number of chars per line restriction?

File: examples/oneapi/cpp/source/decision_forest/df_cls_dense_batch.cpp [hunk 2]
```diff
@@ -56,22 +56,18 @@ int main(int argc, char const *argv[]) {
...
+  const auto result_train = dal::train(df_desc, x_train_table, y_train_table, df::train_result_to_compute::compute_out_of_bag_error);
 
   std::cout << "Variable importance results:" << std::endl
             << result_train.get_var_importance() << std::endl;
 
   std::cout << "OOB error: " << result_train.get_oob_err() << std::endl;
 
   const auto result_infer =
-      dal::infer(df_desc, result_train.get_model(), x_test_table);
+      dal::infer(df_desc, result_train.get_model(), x_test_table, df::infer_result_to_compute::compute_class_labels |
```
Comment (rlnx) [inline examples/oneapi/cpp/source/decision_forest/df_cls_dense_batch.cpp:69]: Seems like line wrapping convention is different for classification and regression examples

**Comment by makart19** [comment]
Renaming train_results_to_compute with error_metrics_to_compute.
Infer_results_to_compute -> results_to_compute.
And returned them back into descriptor (due to only one results_to_compute available now).
Also remove std::uint64_t from input and rewrote operators correspondingly.
Please see request https://github.com/oneapi-src/oneDAL/pull/799

**Comment by napetrov** [comment]
@makart19  - i assume we don't need this PR?

**Comment by napetrov** [comment]
@makart19 - do we need this PR or it can be removed?

**Pull request closed without merging**